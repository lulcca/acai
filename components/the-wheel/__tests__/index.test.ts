import { beforeEach, describe, expect, it, test } from 'vitest';
import TheWheel from '@/components/the-wheel/index.vue';
import { mountSuspended } from '@nuxt/test-utils/runtime';

describe('Components > TheWheel', () => {
  let wrapper: ReturnType<typeof mountSuspended>;

  beforeEach(async () => wrapper = await mountSuspended(TheWheel));

  it('should match snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should render the participants input correctly', () => {
    const textarea = wrapper.find('textarea');
    expect(textarea.exists()).toBe(true);
    expect(textarea.attributes('placeholder')).toBe('Ex.: Nome, Nome, Nome, ...');
  });

  it('should render the clear and spin buttons', () => {
    const buttons = wrapper.findAll('button');
    expect(buttons.length).toBe(2);
    expect(buttons[0].text()).toMatch(/Limpar/i);
    expect(buttons[1].text()).toMatch(/Girar/i);
  });

  it('should update name_array correctly when names input is changed', async () => {
    const namesInput = wrapper.find('textarea');
    await namesInput.setValue('João, Maria, João');
    const nameArray = wrapper.vm.name_array;
    expect(nameArray.length).toBe(2);
    expect(nameArray[0].content).toBe('João');
    expect(nameArray[1].content).toBe('Maria');
    expect(nameArray[0].size).toBe(2);
  });

  test('should clear the participants list when "Limpar" is clicked', async () => {
    const namesInput = wrapper.find('textarea');
    await namesInput.setValue('Lucca, Victor');
    const clearButton = wrapper.find('button', { name: 'Limpar' });
    await clearButton.trigger('click');
    const nameArray = wrapper.vm.name_array;
    expect(nameArray.length).toBe(0);
  });
});
