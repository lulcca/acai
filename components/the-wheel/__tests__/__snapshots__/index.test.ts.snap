// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Components > TheWheel > should match snapshot 1`] = `
"<section id="roleta" class="gap-10 grid grid-cols-1 grid-rows-[auto_1fr] h-screen place-content-start place-items-start w-full">
  <h2 class="font-major text-6xl uppercase"> Roleta </h2>
  <div class="gap-20 grid grid-cols-[2fr_3fr] h-full w-full">
    <div class="gap-5 grid items-start content-start"><label>Participantes</label>
      <div class="relative flex"><textarea rows="10" placeholder="Ex.: primeiro nome, segundo nome, ..." class="relative block w-full disabled:cursor-not-allowed disabled:opacity-75 focus:outline-none border-0 form-textarea rounded-md placeholder-gray-400 dark:placeholder-gray-500 text-base px-3.5 py-2.5 shadow-sm bg-white dark:bg-gray-900 text-gray-900 dark:text-white ring-1 ring-inset ring-gray-300 dark:ring-gray-700 focus:ring-2 focus:ring-primary-500 dark:focus:ring-primary-400 resize-none" value=""></textarea></div>
      <div class="flex justify-between"><button type="button" class="focus:outline-none focus-visible:outline-0 disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:cursor-not-allowed aria-disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-base gap-x-2.5 px-3.5 py-2.5 hover:bg-primary-50 disabled:bg-transparent aria-disabled:bg-transparent dark:hover:bg-primary-950 dark:disabled:bg-transparent dark:aria-disabled:bg-transparent focus-visible:ring-2 focus-visible:ring-inset focus-visible:ring-primary-500 dark:focus-visible:ring-primary-400 text-fuchsia-900 dark:text-fuchsia-100 inline-flex items-center">
          <!--v-if--> Limpar
          <!--v-if-->
        </button><button type="button" class="focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:cursor-not-allowed aria-disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-base gap-x-2.5 px-3.5 py-2.5 shadow-sm disabled:bg-primary-500 aria-disabled:bg-primary-500 dark:disabled:bg-primary-400 dark:aria-disabled:bg-primary-400 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-primary-500 dark:focus-visible:outline-primary-400 bg-fuchsia-900 dark:bg-fuchsia-100 text-fuchsia-100 dark:text-fuchsia-900 hover:bg-primary dark:hover:bg-primary inline-flex items-center">
          <!--v-if--> Girar
          <!--v-if-->
        </button></div>
    </div>
    <figure class="grid justify-items-center w-full h-3/4">
      <div class="flex h-full relative rounded-full w-max shadow-2xl">
        <div class="absolute flex justify-center top-0 w-full z-10"><span class="iconify i-tdesign:arrow-triangle-down-filled bg-gray-100 h-16 w-16" aria-hidden="true"></span></div><svg aria-label="Roleta de Descontos" class="relative w-full h-[260px] sm:h-[300px] md:h-[380px] lg:h-full" viewBox="-200 -200 400 400">
          <g style="transform: rotate(0deg);"></g>
        </svg>
      </div>
    </figure>
  </div>
</section>"
`;
